libp2p/utility.nim

```diff
@@ -65,3 +65,16 @@ when defined(libp2p_agents_metrics):
   const
     KnownLibP2PAgents* {.strdefine.} = "nim-libp2p"
     KnownLibP2PAgentsSeq* = KnownLibP2PAgents.safeToLowerAscii().tryGet().split(",")
+
+template tryException*(body: untyped): untyped =
+  block:
+    var res: type(body)
+    when defined(nimHasWarnBareExcept):
+      {.push warning[BareExcept]:off.}
+    try:
+      res = body
+    except CatchableError as exc: raise exc
```

Comment:
Maybe we should add a
`except Defect as exc: raise exc`
Too, to avoid catching defects

Reply:
And then making `except Exception: discard`?

Reply:
I'd rather assert on this unexpected case than swallow it